### PR TITLE
Feat testcases

### LINKED ISSUE #550: Pressing 'Unregister' on Day 6 example breaks app

### PR DESCRIPTION
**Pull request by eneufeld**
Fixes #550

### CODE REVIEW

File: src/path.util.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { JsonSchema } from './models/jsonSchema';
-const PATH_SEGMENTS_TO_IGNORE = ['#', 'properties'];
+// const PATH_SEGMENTS_TO_IGNORE = ['#', 'properties'];
 
 export const getValuePropertyPair = (instance: any, path: string):
   {instance: Object, property: string} => {
-  const validPathSegments =
-      path.split('/').filter(subPath => PATH_SEGMENTS_TO_IGNORE.indexOf(subPath) === -1);
+  const validPathSegments = toDataPathSegments(path);
+      // path.split('/').filter(subPath => PATH_SEGMENTS_TO_IGNORE.indexOf(subPath) === -1);
```
Comment (edgarmueller) [inline src/path.util.ts]: Remove commented code.

File: src/path.util.ts [hunk 2]
```diff
@@ -21,9 +21,30 @@ export const getValuePropertyPair = (instance: any, path: string):
         decodeURIComponent(validPathSegments[validPathSegments.length - 1]) : undefined
   };
 };
-
-export const toDataPath = (path: string): string =>
-  path.split('/').filter(subPath => PATH_SEGMENTS_TO_IGNORE.indexOf(subPath) === -1).join('/');
+const toDataPathSegments = (path: string): Array<string> => {
+  const segments = path.split('/');
+  const startFromRoot = segments[0] === '#' || segments[0] === '';
+  return segments.filter((segment, index) => {
```
Comment (edgarmueller) [inline src/path.util.ts]: Rather first check for `startFromRoot` and formulate simplified filter expressions.

File: src/renderers/layouts/vertical.layout.ts [hunk 3]
```diff
@@ -2,29 +2,49 @@ import { UISchemaElement, VerticalLayout } from '../../models/uischema';
...
+      });
+    }
     this.appendChild(div);
     return this;
   }
-   dispose(): void {
+  dispose(): void {
     // Do nothing
   }
+  notify(type: RUNTIME_TYPE): void {
```
Comment (edgarmueller) [inline src/renderers/layouts/vertical.layout.ts]: The code of `notify` seems to be the same in all the latter cases. Can this be extracted somehow?

File: test/text.control.test.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+  renderer.setDataService(new DataService(data));
+  renderer.setDataSchema(schema);
+  renderer.setUiSchema({type: 'Control', scope: {$ref: '#/properties/name'}} as ControlElement);
+  const result = renderer.render();
+  const input = <HTMLInputElement>result.children[1];
+  input.value = 'Bar';
+  input.oninput(null);
+  t.is(data.name, 'Bar');
+});
+// TODO If I add console log, then I see that the value is set, but cannot verify
```
Comment (edgarmueller) [inline test/text.control.test.ts]: Can we track this as an issue?

Comment (eneufeld) [inline test/text.control.test.ts]: i will open one for this

File: src/renderers/layouts/vertical.layout.ts (hunk 3 above)
Comment (eneufeld) [inline src/renderers/layouts/vertical.layout.ts]: we could create a super class for this or a util, what would you prefer?

File: src/path.util.ts (hunk 1 above)
Comment (eneufeld) [inline src/path.util.ts]: will do

File: src/path.util.ts (hunk 2 above)
Comment (eneufeld) [inline src/path.util.ts]: ok, will look at this again

File: src/renderers/layouts/vertical.layout.ts (hunk 3 above)
Comment (edgarmueller) [inline src/renderers/layouts/vertical.layout.ts]: I would go for the util

**Pull request merged**